ui/app/src/state/epics/preview.ts

```diff
@@ -141,6 +145,30 @@ export default [
         setStoredShowToolsPanel(uuid, state.user.username, state.preview.showToolsPanel);
       }),
       ignoreElements()
-    )
+    ),
   // endregion
+  // region pushIcePanelPage
+  (action$, state$) =>
+    action$.pipe(
+      ofType(pushIcePanelPage.type),
+      withLatestFrom(state$),
+      tap(([{ payload }, state]) => {
+        if (payload) {
+          const uuid = state.sites.byId[state.sites.active].uuid;
+          setStoredICEToolsPanelPage(uuid, state.user.username, payload);
+        }
+      }),
+      ignoreElements()
+    ),
+  // endregion
+  (action$, state$) =>
+    action$.pipe(
+      ofType(popIcePanelPage.type),
+      withLatestFrom(state$),
+      tap(([, state]) => {
+        const uuid = state.sites.byId[state.sites.active].uuid;
+        removeStoredICEToolsPanelPage(uuid, state.user.username);
```

Comment:
You should be checking if the stack is empty before removing the store data.
See `popToolsPanelPage`